src/main/java/com/mate/onlinebookstore/onlinebookstore/service/impl/BookServiceImpl.java

```
@@ -34,4 +35,20 @@ public BookDto findById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException(String.format("Book with id %s not found", id))));
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+        if (findById(id) != null) {
+            Book newBook = bookMapper.toModel(bookRequestDto);
+            newBook.setId(id);
+            bookRepository.save(newBook);
+            return bookMapper.toDto(newBook);
+        }
+        return null;
```

I'd rather throw an exception if book is not found, because by returning OK status user thinks that everything is OK, but actually his update is not applied.